dragonchain/lib/database/redisearch.py

```diff
@@ -299,23 +304,64 @@ def generate_indexes_if_necessary() -> None:
-    # Create block index
-    _log.info("Creating block indexes")
-    _generate_block_indexes()
-    # Create indexes for transactions
-    _log.info("Creating transaction indexes")
-    _generate_transaction_indexes()
-    # Create smart contract index
-    _log.info("Creating smart contract indexes")
-    _generate_smart_contract_indexes()
-    # Mark index generation as complete
-    _log.info("Marking redisearch index generation complete")
-    redisearch_redis_client.delete(BLOCK_MIGRATION_KEY)
-    redisearch_redis_client.delete(TXN_MIGRATION_KEY)
-    redisearch_redis_client.set(INDEX_GENERATION_KEY, "a")
+
+    if needs_l5_generation:
+        # Create L5 verification indexes
+        _generate_l5_verification_indexes()
+        # Mark index generation as complete
+        redisearch_redis_client.delete(L5_BLOCK_MIGRATION_KEY)
+        redisearch_redis_client.set(INDEX_L5_VERIFICATION_GENERATION_KEY, "a")
+
+    if needs_generation:
+        # Create block index
+        _log.info("Creating block indexes")
+        _generate_block_indexes()
+        # Create indexes for transactions
+        _log.info("Creating transaction indexes")
+        _generate_transaction_indexes()
+        # Create smart contract index
+        _log.info("Creating smart contract indexes")
+        _generate_smart_contract_indexes()
+        # Mark index generation as complete
+        _log.info("Marking redisearch index generation complete")
+        redisearch_redis_client.delete(BLOCK_MIGRATION_KEY)
+        redisearch_redis_client.delete(TXN_MIGRATION_KEY)
+        redisearch_redis_client.set(INDEX_GENERATION_KEY, "a")
+
+
+def _generate_l5_verification_indexes() -> None:
+    client = _get_redisearch_index_client(Indexes.verification.value)
+    try:
+        client.create_index(
+            [
+                redisearch.NumericField("block_id", sortable=True),
+                redisearch.NumericField("prev_id", sortable=True),
+                redisearch.NumericField("timestamp", sortable=True),
+                redisearch.TextField("dc_id"),
+            ]
+        )
+    except redis.exceptions.ResponseError as e:
+        if not str(e).startswith("Index already exists"):  # We don't care if index already exists
+            raise
+    _log.info("Listing all blocks in storage")
+    block_paths = storage.list_objects("BLOCK/")
+    pattern = re.compile(r"BLOCK\/([0-9]+)-([Ll])5(.*)$")
+    for block_path in block_paths:
+        if LEVEL == "1" and BROADCAST_ENABLED and re.search(pattern, block_path):
+            if not client.redis.sismember(L5_BLOCK_MIGRATION_KEY, block_path):
+                raw_block = storage.get_json_from_object(block_path)
+                block = l5_block_model.new_from_at_rest(raw_block)
+                _log.info("STORING L5 VERIFICATION:")
+                _log.info(block.export_as_search_index())
+                put_document(Indexes.verification.value, block_path.split("/")[1], block.export_as_search_index())
```

Comment:
what are the contents of `block_path.split("/")[1]` ?

Remember that regardless of the index name, the document name here must be _globally_ unique in the redisearch

Reply:
the block path in this case is `BLOCK/<block_id>-L5-<l5_dc_id>` so I only want this part `<block_id>-L5-<l5_dc_id>`